client.go

```diff
@@ -29,14 +28,325 @@ func main() {
 	config.Cluster.Cloud.Token = "YOUR_CLUSTER_DISCOVERY_TOKEN"
 	config.Stats.Enabled = true
 	config.Stats.Period = types.Duration(time.Second)
-
 	client, err := hazelcast.StartNewClientWithConfig(ctx, config)
 	if err != nil {
 		panic(err)
 	}
 	defer client.Shutdown(ctx)
-	log.Println("Connection Successful!")
-	log.Println("Now, `map` will be filled with random entries.")
+	fmt.Println("Connection Successful!")
+
+	// Uncomment the example you want to run.
+	mapExample(client)
+	//sqlExample(client)
+	//jsonSerializationExample(client)
+	//nonStopMapExample(client)
+}
+
+// mapExample shows how to work with Hazelcast maps.
+// It simply puts a bunch of entry to map.
+func mapExample(client *hazelcast.Client) {
+	ctx := context.Background()
+	cities, err := client.GetMap(ctx, "cities")
+	if err != nil {
+		panic(err)
+	}
+	cities.Put(ctx, "1", model.CityAsJson("United Kingdom", "London", 9_540_576))
+	cities.Put(ctx, "2", model.CityAsJson("United Kingdom", "Manchester", 2_770_434))
+	cities.Put(ctx, "3", model.CityAsJson("United States", "New York", 19_223_191))
+	cities.Put(ctx, "4", model.CityAsJson("United States", "Los Angeles", 3_985_520))
+	cities.Put(ctx, "5", model.CityAsJson("Turkey", "Ankara", 5_309_690))
+	cities.Put(ctx, "6", model.CityAsJson("Turkey", "Istanbul", 15_636_243))
+	cities.Put(ctx, "7", model.CityAsJson("Brazil", "Sao Paulo", 22_429_800))
+	cities.Put(ctx, "8", model.CityAsJson("Brazil", "Rio de Janeiro", 13_634_274))
+	mapSize, err := cities.Size(ctx)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("'cities' map now contains %d entries.\n", mapSize)
+	fmt.Println("--------------------")
+}
+
+// sqlExample shows how to work with Hazelcast SQL queries.
+// It maps maps to database tables and runs queries on these maps.
+func sqlExample(client *hazelcast.Client) {
+	sqlService := client.SQL()
+	// Separate examples that shows SQL usage of Hazelcast Go Client
+	createMappingForCapitals(sqlService)
+	clearCapitals(sqlService)
+	populateCapitals(sqlService)
+	selectAllCapitals(sqlService)
+	selectCapitalNames(sqlService)
+}
+
+func createMappingForCapitals(sqlService sql.Service) {
+	// See: https://docs.hazelcast.com/hazelcast/5.1/sql/mapping-to-maps
+	fmt.Println("Creating a mapping...")
+	mappingQuery := fmt.Sprintf(`
+        CREATE OR REPLACE MAPPING capitals TYPE IMap
+		OPTIONS (
+			'keyFormat' = 'varchar',
+			'valueFormat' = 'varchar'
+		)
+	`)
+	ignored, err := sqlService.Execute(context.Background(), mappingQuery)
+	if err != nil {
+		panic(err)
+	}
+	defer ignored.Close()
+	fmt.Println("The mapping has been created successfully.")
+	fmt.Println("--------------------")
+}
+
+func clearCapitals(sqlService sql.Service) {
+	fmt.Println("Deleting data via SQL...")
+	ignored, err := sqlService.Execute(context.Background(), "DELETE FROM capitals")
+	if err != nil {
+		panic(err)
+	}
+	defer ignored.Close()
```

Comment:
Defering closing should be done just after the `sqlService.Execute` otherwise in the case of panic it won't be closed because of the order in the stack. Same comment for rest of them.